### PR TITLE
[Enhancement Shaman] Updates

### PR DESCRIPTION
**Pull request by Seriousnes**
### Description

* Malestrom Weapon tracker has been rewritten to provide more accurate tracking
* Added maelstrom usage details table
* Added maelstrom efficiency table

### Testing
- Test report URL: `/report/WcFTvqQHZJhKwNzp/19-Mythic+Scalecommander+Sarkareth+-+Wipe+16+(5:39)/Seriousnes/standard/statistics`
- Screenshot(s):

Usage details
image

Efficiency
image

### CODE REVIEW

File: src/analysis/retail/shaman/enhancement/modules/normalizers/EventLinkNormalizer.tsx [hunk 1]
```diff
@@ -75,13 +65,47 @@ const chainLightningDamageLink: EventLink = {
   anyTarget: true,
 };
 
+const maelstromWeaponSenderLink: EventLink = {
```
Comment (emallson) [inline src/analysis/retail/shaman/enhancement/modules/normalizers/EventLinkNormalizer.tsx]: typo

```suggestion
const maelstromWeaponSpenderLink: EventLink = {
```

File: src/analysis/retail/shaman/enhancement/modules/dragonflight/Tier30.tsx [hunk 2]
```diff
@@ -99,31 +99,36 @@ export default class Tier30 extends Analyzer {
           <>
             <SpellLink spell={SPELLS.VOLCANIC_STRENGTH_TIER_BUFF} />
             <ul>
-              <li>
+              <li key="phys">
```
Comment (emallson) [inline src/analysis/retail/shaman/enhancement/modules/dragonflight/Tier30.tsx]: just as a note: the `key` prop is only needed when using arrays as JSX. when you've spelled it out like this the `key` prop doesn't add anything. not something that needs fixing, but a note

File: src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponSpenders.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    const primordialWaveLightningBolt = event._linkedEvents?.find(
+      (le) => le.relation === PRIMORIDAL_WAVE_END_LINK,
+    )?.event;
+    if (primordialWaveLightningBolt) {
+      const spent = this.maelstromWeaponTracker.current;
+      this.maelstromSpendWithPrimordialWave +=
+        spent === 0 ? 10 : this.maelstromWeaponTracker.current;
+      const damageEvents: DamageEvent[] = primordialWaveLightningBolt._linkedEvents
+        ?.filter((le) => le.relation === LIGHTNING_BOLT_PRIMORDIAL_WAVE_LINK)
+        .map((le) => le.event) as DamageEvent[];
```
Comment (emallson) [inline src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponSpenders.tsx]: note that we have the `GetRelatedEvents` helper for cases like this (it is a little more awkward for cases with a single event like PW LB)

```suggestion
      const damageEvents: DamageEvent[] = GetRelatedEvents(primordialWaveLightningBolt, LIGHTNING_BOLT_PRIMORDIAL_WAVE_LINK) as DamageEvent[];
```

File: src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponTracker.tsx [hunk 4]
```diff
@@ -20,7 +35,7 @@ const MaelstromWeaponResource: Resource = {
   url: 'maelstrom_weapon',
 };
 
-class MaelstromWeaponTracker extends ResourceTracker {
+export default class extends ResourceTracker {
```
Comment (emallson) [inline src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponTracker.tsx:38]: minor style thing: setting the class name here can help in debugging to make it clearer what type an object is. not required to merge, though

**Review by emallson — APPROVED** [review]
sorry for the late review!

some minor comments but nothing that actually *requires* intervention before merging, so i'll merge sometime tomorrow unless i hear something from you about wanting to address any comments

File: src/analysis/retail/shaman/enhancement/modules/dragonflight/Tier30.tsx (hunk 2 above)
Comment (Seriousnes) [reply src/analysis/retail/shaman/enhancement/modules/dragonflight/Tier30.tsx]: I'll remove them, I think I was a bit overzealous when trying to fix the missing key properties I just added it to every `<li>` I've used in the entire enhancement module

File: src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponSpenders.tsx (hunk 3 above)
Comment (Seriousnes) [reply src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponSpenders.tsx]: Done

File: src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponTracker.tsx (hunk 4 above)
Comment (Seriousnes) [reply src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponTracker.tsx:38]: I'll keep this in mind for next time

**Review by emallson — APPROVED** [review]

**Pull request merged**